std/special/bootstrap.zig

```diff
@@ -123,23 +125,30 @@ inline fn callMain() u8 {
         },
         builtin.TypeId.Int => {
             if (@typeOf(root.main).ReturnType.bit_count != 8) {
-                @compileError("expected return type of main to be 'u8', 'noreturn', 'void', or '!void'");
+                @compileError(compile_err);
             }
             return root.main();
         },
         builtin.TypeId.ErrorUnion => {
-            root.main() catch |err| {
-                std.debug.warn("error: {}\n", @errorName(err));
-                if (builtin.os != builtin.Os.zen) {
-                    if (@errorReturnTrace()) |trace| {
-                        std.debug.dumpStackTrace(trace.*);
+            const PayloadType = @typeOf(root.main).ReturnType.Payload;
+            // In this case the error should include the payload type
+            const payload_err = "expected return type of main to be 'u8', 'noreturn', 'void', '!void', or '!u8', found '!" ++ @typeName(PayloadType) ++ "'";
```

Comment:
why not use `compile_err`?

Reply:
Because `compile_err` uses the name of the return type, which is fine when it isn't an error union since then it looks like `foo.bar.ErrorUnion!T` so I use a different variable when it is an error union. I could make `compile_err` a `var` and reassign it.

Reply:
I just don't like the duplication, how about:
```zig
const compile_err_prefix = "expected return type of main to be 'u8', 'noreturn', 'void', '!void', or '!u8'"
...
// in the payload case
const compile_err = compile_err_prefix ++ ", found '!" ++ @typeName(PayloadType) ++ "'";
...
// in the other cases
const compile_err = compile_err_prefix ++ ", found '" ++ @typeName(@typeOf(root.main).ReturnType) ++ "'";
```